xarray/plot/dataarray_plot.py

```diff
@@ -148,48 +149,45 @@ def _infer_line_data(
     return xplt, yplt, hueplt, huelabel
 
 
-def _infer_plot_dims(
-    darray: DataArray,
-    dims_plot: MutableMapping[str, Hashable],
-    default_guess: Iterable[str] = ("x", "hue", "size"),
-) -> MutableMapping[str, Hashable]:
+def _infer_line_data2(
```

Comment:
```suggestion
def _infer_scatter_data(
```

Isn't this only used for scatter at the moment?

Reply:
Only used for scatter at the moment (see #7174), it mostly does broadcasting and reshaping for plt.scatter now. What do we call these types of functions, `_prepare_plot1d_data`?

Reply:
Done.